### PR TITLE
feat(atsu): Datalink status page and Max uplink delay page

### PR DESCRIPTION
**Pull request by svengcz**
## Summary of Changes
This PR adapts the datalink status page to the Hoppie integration and adapts it to MSN7560 layout.
Additionally are some visualizations in the ATC pages fixed to be MSN7560 compliant
The Max Uplink Delay page is also integrated.

### Updated non-editable entries in MCDU pages
grafik
grafik
grafik

### Fixed visualization issues w.r.t MSN7560
grafik

### Max uplink delay (logged in)
grafik

### Max uplink delay (logged off)
grafik

### Datalink status without Hoppie connection
grafik

### Datalink status with Hoppie connection
grafik

### Printed datalink page
grafik

## References

A320 FCOM documentation for MSN7560

## Additional context

Discord username (if different from GitHub): Sven [de en]

## Testing instructions

- tests of non-editable 
  - keep INIT-page empty
    - AOC/DEPART REQ should show ------ for the callsign and from/to
    - Atc/Connection status should show ---- for current and next ATC
    - Atc/Notification should show ---- for the callsign
  - fill out INIT-page
    - AOC/DEPART REQ should be filled for callsign and from/to
    - Atc/Connection should be the same as above
    - Atc/Notification should show the callsign
- tests of datalink status
  - open the Datalink Status page
    - VHF3 is supposed to show AOC Only
  - activate Hoppie connection via EFB
    - Datalink Status should show without a page reload that AOC/ATC is shown below VHF3
    - deactivation of Hoppie returns to AOC Only
  - All other com channels are not installed
- test print function of datalink status
- Connection Status page
  - check if a handover from one station to an other shows the following information
    - ACTIVE ATC: station that sent the handover
    - NEXT ATC: the following station
    - After an accepted logon of the next station:
    - ACTIVE ATC: old NEXT ATC
    - NEXT ATC: empty
- MAX UPLINK DELAY in CONNECTION
  - Not logged on: show ---- for ACTIVE ATC
  - Logged on: show station in green letters
  - After a logon or handoff is MAX UPLINK DELAY NONE
  - can be set via [1] (is reset after a station change)
  - not editable max uplink delay if no station is active

[1] 
grafik

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_DatalinkStatus.js [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
+            }
+        }
+
+        mcdu.refreshPageCallback = () => {
+            updateView();
+        };
+        SimVar.SetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number", 1);
+
+        let vhf3Mode = "AOC ONLY";
+        if (SimVar.GetSimVarValue('L:A32NX_HOPPIE_ACTIVE', 'number') === 1) {
```
Comment (2hwk) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_DatalinkStatus.js:19]:
```suggestion
        if (SimVar.GetSimVarValue("L:A32NX_HOPPIE_ACTIVE", "number") === 1) {
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_DatalinkStatus.js [hunk 2]
```diff
@@ -1,27 +1,62 @@
-class CDUDatalinkStatus {
+class CDUAtsuDatalinkStatus {
     static ShowPage(mcdu) {
         mcdu.clearDisplay();
+        mcdu.page.Current = mcdu.page.ATSUDatalinkStatus;
+        mcdu.activeSystem = 'ATSU';
```
Comment (2hwk) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_DatalinkStatus.js:5]:
```suggestion
        mcdu.activeSystem = "ATSU";
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_Menu.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class CDUAtsuMenu {
     static ShowPage(mcdu) {
         mcdu.clearDisplay();
+        mcdu.page.Current = mcdu.page.ATSUMenu;
         mcdu.activeSystem = 'ATSU';
```
Comment (2hwk) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATSU_Menu.js:5]:
```suggestion
        mcdu.activeSystem = "ATSU";
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATC_ConnectionStatus.js [hunk 4]
```diff
@@ -12,16 +24,16 @@ class CDUAtcConnectionStatus {
             store["disconnectAvail"] = false;
         }
 
-        let nextStation = "----";
+        let nextStation = "-----------";
         if (mcdu.atsuManager.atc.nextStation() !== '') {
```
Comment (2hwk) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATC_ConnectionStatus.js:28]:
```suggestion
        if (mcdu.atsuManager.atc.nextStation() !== "") {
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATC_ConnectionStatus.js [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+        }
+
+        mcdu.refreshPageCallback = () => {
+            updateView();
+        };
+        SimVar.SetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number", 1);
+
+        let currentStation = "-----------[color]white";
         let atcDisconnect = "DISCONNECT\xa0[color]cyan";
         if (mcdu.atsuManager.atc.currentStation() !== '') {
```
Comment (2hwk) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATC_ConnectionStatus.js:19]:
```suggestion
        if (mcdu.atsuManager.atc.currentStation() !== "") {
```

[the remark this answers is not in the document]
Comment (2hwk) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_ATC_ConnectionStatus.js:19]: Our style guide/convention is to use double quotes in js. 

(In typescript we have changed to use single quotes, and there are still portions where this is missed sometimes).

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -131,8 +131,6 @@
 1. [AP] Improved LOC ALGIN and ROLL OUT during Autoland - @aguther (Andreas Guther)
 1. [HYD] Simple hydraulic model placeholder for flaps and slats - @Crocket63
 1. [FBW] Fix: do not compensate ground spoilers in C* law - @aguther (Andreas Guther)
-1. [EFB] Added Hoppie configuration to ATSU/ATC page - @svengcz (Sven Czarnian)
```
[the remark this answers is not in the document]
Comment (svengcz) [reply .github/CHANGELOG.md:134]: That was discussed in the discord. I created (5?) entries with my first PR. We agreed to clean it up.

**Review by 2hwk — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #6596 
Tier of Testing  : 1
Date             : 19/02/2022

Testing Process: 
tests of non-editable
keep INIT-page empty
AOC/DEPART REQ should show ------ for the callsign and from/to - OK
Atc/Connection status should show ---- for current and next ATC - OK
Atc/Notification should show ---- for the callsign - OK

fill out INIT-page
AOC/DEPART REQ should be filled for callsign and from/to - OK
Atc/Connection should be the same as above - OK
Atc/Notification should show the callsign - OK

tests of datalink status
open the Datalink Status page 
VHF3 is supposed to show AOC Only - OK

activate Hoppie connection via EFB
Datalink Status should show without a page reload that AOC/ATC is shown below VHF3 - OK
deactivation of Hoppie returns to AOC Only - OK

All other com channels are not installed - OK

test print function of datalink status - OK

Connection Status page
check if a handover from one station to an other shows the following information
ACTIVE ATC: station that sent the handover - OK
NEXT ATC: the following station - OK
After an accepted logon of the next station:
ACTIVE ATC: old NEXT ATC - OK
NEXT ATC: empty - OK

MAX UPLINK DELAY in CONNECTION
Not logged on: show ---- for ACTIVE ATC - OK
Logged on: show station in green letters - OK
After a logon or handoff is MAX UPLINK DELAY NONE - OK
can be set via [1] (is reset after a station change) - OK
not editable max uplink delay if no station is active - OK

Negatives:
N/A

Testing Results: 
Passed

Conclusions: 
Nice work 🚀

**Pull request merged**